.github/workflows/docker.yml

```diff
@@ -1,34 +1,48 @@
 name: Docker
+
 on:
+  workflow_dispatch: # allows to be run manually
   push:
-    tags: [ "*" ]
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - 'master'
 
 jobs:
-  push_to_registry:
-    name: Push Docker image to Docker hub
+  build_and_push:
+    name: Build and push Docker image to Docker Hub
     runs-on: ubuntu-latest
     steps:
-      - name: Check out the repo
+      - name: Checkout
```

Comment:
I don't quite see why changes like this are necessary. No need to change them back, just recording my bafflement for posterity.